demucs/separate.py

```diff
@@ -144,61 +170,68 @@ def main(opts=None):
-                              num_workers=args.jobs)[0]
+        sources = apply_model(
+            model,
+            wav[None],
+            device=args.device,
+            shifts=args.shifts,
+            split=args.split,
+            overlap=args.overlap,
+            progress=True,
+            num_workers=args.jobs,
+        )[0]
         sources = sources * ref.std() + ref.mean()
 
         if args.mp3:
             ext = "mp3"
         else:
             ext = "wav"
         kwargs = {
-            'samplerate': model.samplerate,
-            'bitrate': args.mp3_bitrate,
-            'clip': args.clip_mode,
-            'as_float': args.float32,
-            'bits_per_sample': 24 if args.int24 else 16,
+            "samplerate": model.samplerate,
+            "bitrate": args.mp3_bitrate,
+            "clip": args.clip_mode,
+            "as_float": args.float32,
+            "bits_per_sample": 24 if args.int24 else 16,
         }
         if args.stem is None:
             for source, name in zip(sources, model.sources):
-                stem = out / args.filename.format(track=track.name.rsplit(".", 1)[0],
-                                                  trackext=track.name.rsplit(".", 1)[-1],
-                                                  stem=name, ext=ext)
+                stem = out / args.filename.format(
+                    track=track.name.rsplit(".", 1)[0],
+                    trackext=track.name.rsplit(".", 1)[-1],
+                    stem=name,
+                    ext=ext,
+                )
                 stem.parent.mkdir(parents=True, exist_ok=True)
                 save_audio(source, str(stem), **kwargs)
         else:
             sources = list(sources)
-            stem = out / args.filename.format(track=track.name.rsplit(".", 1)[0],
-                                              trackext=track.name.rsplit(".", 1)[-1],
-                                              stem=args.stem, ext=ext)
+            stem = out / args.filename.format(
+                track=track.name.rsplit(".", 1)[0],
+                trackext=track.name.rsplit(".", 1)[-1],
+                stem=args.stem,
+                ext=ext,
+            )
             stem.parent.mkdir(parents=True, exist_ok=True)
             save_audio(sources.pop(model.sources.index(args.stem)), str(stem), **kwargs)
-            # Warning : after poping the stem, selected stem is no longer in the list 'sources'
-            other_stem = th.zeros_like(sources[0])
-            for i in sources:
-                other_stem += i
-            stem = out / args.filename.format(track=track.name.rsplit(".", 1)[0],
-                                              trackext=track.name.rsplit(".", 1)[-1],
-                                              stem="no_"+args.stem, ext=ext)
-            stem.parent.mkdir(parents=True, exist_ok=True)
-            save_audio(other_stem, str(stem), **kwargs)
 
```

Comment:
Some users would like to save a file without vocals but not an audio with only vocals, so these codes should be kept.

Reply:
Oh, sorry. I'm new to git so I made a mistake. All I intended was to edit readme.md at the very beginning. I will do a pull request again.